File: src/04.FrontEndReact/04.FrontEndReact.Logics/Modules/MasterData/Countries/DeleteCountry/DeleteCountryCommand.ts
import { DeleteCountryRoute } from "@services/Dto/Modules/MasterData/Countries/DeleteCountry/DeleteCountryRoute";
import { getBackEndApiService } from "@infrastructure/BackEndApi/ConfigureBackEndApi";

export async function sendDeleteCountryCommand(request: { id: string }): Promise<void> {
  await getBackEndApiService().sendRequest({
    resourceUri: DeleteCountryRoute.ResourceUri,
    method: "DELETE",
    query: { id: request.id },
  });
}

File: src/03.FrontEnd/04.FrontEnd.Logics/Modules/MasterData/Countries/DeleteCountry/DeleteCountryCommand.cs
using SolutionTemplate2.Shared.Dto.Modules.MasterData.Countries.DeleteCountry;

namespace SolutionTemplate2.FrontEnd.Logics.Modules.MasterData.Countries.DeleteCountry;

public sealed record DeleteCountryCommand : DeleteCountryRequest, IRequest
{
}

public sealed class DeleteCountryCommandValidator : AbstractValidatorBase<DeleteCountryCommand>
{
    public DeleteCountryCommandValidator()
    {
        Include(new DeleteCountryRequestValidator());
    }
}

public sealed class DeleteCountryCommandHandler(IBackEndApiService backEndApiService)
    : IRequestHandler<DeleteCountryCommand>
{
    public async Task Handle(DeleteCountryCommand request, CancellationToken cancellationToken)
    {
        var restRequest = new RestRequest(DeleteCountryRoute.ResourceUri(request.CountryId), Method.Delete);

        await backEndApiService.SendRequestAsync(restRequest, cancellationToken);
    }
}
